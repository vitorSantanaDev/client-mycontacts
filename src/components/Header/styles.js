import styled, { css } from 'styled-components'

export const Container = styled.header`
	${({ theme }) => css`
		display: flex;
		margin-top: 7.4rem;
		align-items: center;
		flex-direction: column;
		justify-content: center;
		margin-bottom: ${theme.spacings.xlarge};
	`}
`
