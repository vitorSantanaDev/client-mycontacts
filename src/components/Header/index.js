import { Container } from './styles'

import Logo from '../../assets/images/logo-mycontacts.svg'

export default function Header() {
	return (
		<Container>
			<img src={Logo} alt="My contacts" width={201} />
		</Container>
	)
}
