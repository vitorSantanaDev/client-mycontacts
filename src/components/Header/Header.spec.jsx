import React from 'react'
import renderWithTheme from '../../utils/render-with-theme'

import Header from '.'
import { screen } from '@testing-library/react'

describe('<Header />', () => {
	it('should render the header component correctly', () => {
		renderWithTheme(<Header />)

		expect(screen.getByAltText(/My contacts/i)).toBeInTheDocument()
		expect(
			screen.getByPlaceholderText(/Pesquisar contato.../i)
		).toBeInTheDocument()
	})
	it('should match snapshot', () => {
		const { container } = renderWithTheme(<Header />)

		expect(container.firstChild).toMatchSnapshot()
	})
})
