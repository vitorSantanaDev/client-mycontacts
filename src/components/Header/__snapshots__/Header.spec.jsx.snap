// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should match snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-top: 7.4rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-bottom: 4.8rem;
}

<header
  class="c0"
>
  <img
    alt="My contacts"
    src="logo-mycontacts.svg"
    width="201"
  />
</header>
`;
